mods/tnt/init.lua

```diff
@@ -345,6 +345,14 @@ local function tnt_explode(pos, radius, ignore_protection, ignore_on_blast, owne
 	local on_construct_queue = {}
 	basic_flame_on_construct = minetest.registered_nodes["fire:basic_flame"].on_construct
 
+	-- Used to efficiently remove metadata of nodes that were destroyed.
+	-- Metadata is probably sparse, so this may save us some work.
+	local poses_with_meta = minetest.find_nodes_with_meta(p1, p2)
+	local has_meta = {}
+	for _, p in ipairs(poses_with_meta) do
+		has_meta[a:indexp(p)] = true
+	end
+
```

Comment:
```suggestion
	local has_meta = {}
	for _, p in ipairs(minetest.find_nodes_with_meta(p1, p2)) do
		has_meta[a:indexp(p)] = true
	end

```

Reply:
sure, why not. though i don't think it matters much.